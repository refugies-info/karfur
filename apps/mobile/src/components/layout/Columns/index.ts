export { default, ColumnsSpacing } from "./Columns";
