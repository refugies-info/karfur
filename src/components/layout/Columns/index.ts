export { default } from "./Columns";
export { ColumnsSpacing } from "./Columns";
