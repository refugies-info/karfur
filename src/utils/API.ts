import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
