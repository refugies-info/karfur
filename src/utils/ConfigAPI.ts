import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

const dbURL = Config.dbUrl;
const siteSecret = Config.siteSecret;
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});