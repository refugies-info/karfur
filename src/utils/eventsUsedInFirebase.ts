export const events = {
  REACHED_END_CONTENT: "REACHED_END_CONTENT",
  LANGUAGE_VALIDATE: "LANGUAGE_VALIDATE",
  LOCATION_VALIDATE: "LOCATION_VALIDATE",
};
