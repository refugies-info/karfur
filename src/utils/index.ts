export { default as withProps } from "./withProps";
