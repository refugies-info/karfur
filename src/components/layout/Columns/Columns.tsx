import React, { ReactNode } from "react";
import styled from "styled-components/native";
import { FlexItem, getFlexValue } from "../common";
import { isLastChild } from "../../utils";

const ColumnsWrapper = styled.View<{
  horizontalAlign?: string;
  RTLBehaviour: boolean;
  verticalAlign?: string;
}>`
  flex-grow: 1;
  flex-basis: auto;
  flex-direction: ${({ theme, RTLBehaviour }) =>
    RTLBehaviour && theme.i18n.isRTL ? "row-reverse" : "row"};
  ${({ verticalAlign }) => verticalAlign && `align-items: ${verticalAlign}`};
  ${({ horizontalAlign }) =>
    horizontalAlign && `justify-content: ${horizontalAlign}`};
`;

export enum ColumnsSpacing {
  Default = "default",
  NoSpace = "nospace",
}

export interface ColumnsProps {
  children: ReactNode;
  horizontalAlign?: string;
  layout?: string;
  RTLBehaviour?: boolean;
  spacing?: ColumnsSpacing;
  verticalAlign?: string;
}

const Columns = ({
  children,
  horizontalAlign = "flex-start",
  layout = "1",
  RTLBehaviour = false,
  spacing = ColumnsSpacing.Default,
  verticalAlign = "flex-start",
}: ColumnsProps) => (
  <ColumnsWrapper
    horizontalAlign={horizontalAlign}
    RTLBehaviour={RTLBehaviour}
    verticalAlign={verticalAlign}
  >
    {React.Children.map(children, (child, index) => (
      <FlexItem
        flex={getFlexValue(layout, index)}
        // eslint-disable-next-line react/no-array-index-key
        key={index}
        RTLBehaviour={RTLBehaviour}
        marginHorizontal={!isLastChild(children, index) && spacing}
      >
        {child}
      </FlexItem>
    ))}
  </ColumnsWrapper>
);

export default Columns;
