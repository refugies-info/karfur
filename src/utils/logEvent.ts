/* eslint-disable no-console */
import * as Analytics from "expo-firebase-analytics";
import { getEnvironment } from "../libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (
  eventName: FirebaseEvent,
  data: Record<string, any>
) => {
  const { envName, debugModeFirebase } = getEnvironment();
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase === "activated") {
      Analytics.setDebugModeEnabled(true);
      console.log(
        "Environment is dev and debug mode is activated for firebase"
      );
    } else {
      console.log(
        "Environment is dev and debug mode is NOT activated for firebase"
      );
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await Analytics.logEvent(eventName, { ...data, envName });
  }
};
