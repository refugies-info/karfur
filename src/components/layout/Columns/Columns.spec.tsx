import React from "react";
import { Text } from "react-native";
import Columns from "./Columns";
import { render } from "../../utils/tests";

describe("Columns snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = await render(
      <Columns>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
      </Columns>
    );
    expect(test).toMatchSnapshot();
  });

  it("should be able to control flex layout", async () => {
    const test = await render(
      <Columns
        layout="1 2"
        horizontalAlign="space-between"
        verticalAlign="center"
      >
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
        <Text>Test</Text>
      </Columns>
    );
    expect(test).toMatchSnapshot();
  });
});
