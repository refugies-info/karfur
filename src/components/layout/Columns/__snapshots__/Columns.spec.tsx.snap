// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns snapshot test suite should be able to control flex layout 1`] = `
<View
  RTLBehaviour={false}
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  horizontalAlign="space-between"
  style={
    {
      "alignItems": "center",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  verticalAlign="center"
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 2,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 2,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;

exports[`Columns snapshot test suite should render without bug 1`] = `
<View
  RTLBehaviour={false}
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  horizontalAlign="flex-start"
  style={
    {
      "alignItems": "flex-start",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 1,
      "justifyContent": "flex-start",
    }
  }
  verticalAlign="flex-start"
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "marginRight": 8,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
